src/libraries/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFileValidFullName_Unix.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text;
+
+namespace System.IO.Compression
+{
+    public static partial class ZipFileExtensions
+    {
+        internal static string SanitizeZipFilePath(string zipPath)
+        {
+            StringBuilder builder = new StringBuilder(zipPath);
+            for (int i = 0; i < zipPath.Length; i++)
+            {
+                if ((int)builder[i] == 0)
+                {
+                    builder[i] = '_';
+                }
+            }
+            return builder.ToString();
```

Comment:
This is shorter and more efficient (eg., no allocations in happy path)
```suggestion
            return zipPath.Replace('\0', '_')
```

although, do we have any evidence that nulls are ever found in file paths in valid real world zip files? the situation with invalid Windows file characters is quite different, because that occurs when you extract from a valid archive made from a Unix system. But null characters are invalid everywhere.

Reply:
Eh, I think it's fine to leave the null sanitization as it's a natural extension and it's only a matter of time before someone somehow encounters this.

Reply:
The tests included null characters so I didn't leave sanitization out for null characters. Also good for completeness I guess